src/cdomains/intDomain.ml

```diff
@@ -1530,10 +1530,33 @@ module Enums : S = struct
         ) Invariant.none ns
 end
 
+module type IntDomTupleConf = sig
+  val def_exc: unit -> bool
+  val interval: unit -> bool
+  val cinterval: unit -> bool
+  val enums: unit -> bool
+end
+
+module IntDomTupleStdConf: IntDomTupleConf = struct
+  let lookup s = get_bool @@ "ana.int."^s
+  let def_exc () = lookup "def_exc"
+  let interval () = lookup "interval"
+  let cinterval () = lookup "cinterval"
+  let enums () = lookup "enums"
+end
+
+module IntDomTupleInvariantConf: IntDomTupleConf = struct
+  let lookup s = get_bool @@ "ana.int."^s
+  let def_exc () = true
+  let interval () = lookup "interval"
+  let cinterval () = lookup "cinterval"
+  let enums () = lookup "enums"
+end
+
 (* The above IntDomList has too much boilerplate since we have to edit every function in S when adding a new domain. With the following, we only have to edit the places where fn are applied, i.e., create, mapp, map, map2. *)
-module IntDomTuple = struct
+module IntDomTupleFunc (M: IntDomTupleConf) = struct
   include Printable.Std (* for default invariant, tag, ... *)
-
+  module Conf = M
```

Comment:
remove?

Reply:
The `Conf` is used in the conversion function back to the regular `IntDomTuple` to check whether the `DefExc.t` value should be retained or not.
https://github.com/goblint/analyzer/blob/e5333b75677a22989c534cccf996999aa17aa12c/src/cdomains/intDomain.ml#L1690

Reply:
Ah, didn't see.